src/resources/SecurityCache/SecurityCacheInterfaces.ts

```diff
@@ -128,9 +129,21 @@ export interface SecurityCacheListOptions {
     securityProviderId?: string;
     states?: SecurityCacheStateOptions;
     to?: string;
-    usePageModel?: boolean;
 }
 
+export interface SecurityCacheListOptionsPageModel extends SecurityCacheListOptions {
+    /** Model structure will change and include total entries/pages **/
+    usePageModel: boolean;
+}
+
+export type SecurityCacheListReturnType<T> = T extends SecurityCacheListOptions
+    ? DetailedSecurityCacheMemberModel[]
+    : T extends SecurityCacheListOptionsPageModel
+    ? PageModel<DetailedSecurityCacheMemberModel>
+    : never;
```

Comment:
I never saw a type like this, At first glance it seems a bit complex. Do you know how this type pattern is called ? I would like to read a bit about it.

Reply:
@dmgauthier is `Conditional Types` https://www.typescriptlang.org/docs/handbook/2/conditional-types.html

Reply:
I agree with @dmgauthier, a ternary within a ternary is always complicated to read and maintain at the same time. Maybe the types could be broken down a bit more interesting?

Reply:
I was gonna say that it's a common pattern, but I finally decided that no, it ain't ok.

Here, the pattern is not doing what it supposed to do I reckon: `PageModel<DetailedSecurityCacheMemberModel>` is 'unattainable' because `SecurityCacheListOptionsPageModel extends SecurityCacheListOptions`. Thus, the first 'if' (i.e. does `T extends SecurityCacheListOptions`?) will be true before the second is elevated, or if it's false, the second will always be false.

To solve the issue without changing the pattern being used:
```suggestion
export type SecurityCacheListReturnType<T> = T extends SecurityCacheListOptionsPageModel
    ? PageModel<DetailedSecurityCacheMemberModel>
    : T extends SecurityCacheListOptions
    ? DetailedSecurityCacheMemberModel[]
    : never;
```

What I think we could do to make it easier for the maintainers:
```suggestion
export type SecurityCacheListReturnType<
    T extends SecurityCacheListOptions
> = T extends SecurityCacheListOptionsPageModel
    ? PageModel<DetailedSecurityCacheMemberModel>
    : DetailedSecurityCacheMemberModel[];
```

The `never` is induced by the `< T extends SecurityCacheListOptions>` that enforces the use of `SecurityCacheListOption for the use of this type and that avoid the nested ternary.